pkg/profile/helm.go

```diff
@@ -0,0 +1,83 @@
+package profile
+
+import (
+	"path"
+	"strings"
+
+	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (p *Profile) makeHelmRepository(url string, name string) *sourcev1.HelmRepository {
+	return &sourcev1.HelmRepository{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      p.makeHelmRepoName(name),
+			Namespace: p.subscription.ObjectMeta.Namespace,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       sourcev1.HelmRepositoryKind,
+			APIVersion: sourcev1.GroupVersion.String(),
+		},
+		Spec: sourcev1.HelmRepositorySpec{
+			URL: url,
+		},
+	}
+}
+
+func (p *Profile) makeHelmRepoName(name string) string {
+	repoParts := strings.Split(p.subscription.Spec.ProfileURL, "/")
+	repoName := repoParts[len(repoParts)-1]
+	return join(p.subscription.Name, repoName, name)
+}
+
+func (p *Profile) makeHelmRelease(artifact profilesv1.Artifact, repoPath string) *helmv2.HelmRelease {
+	var helmChartSpec helmv2.HelmChartTemplateSpec
+	if artifact.Path != "" {
```

Comment:
On or the other **must** be defined, right..?

Reply:
just copied this across, but yeah. I think its validated in the profile definition? Idr